dell/xps/17-9710/intel/default.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, pkgs, ... }: {
+  imports = [
+    ../../../../common/cpu/intel
+    ../../../../common/pc/laptop
+    ../../../../common/pc/ssd
+  ];
+
+  # patch the kernel to enable sound
+  # Tested with 5.15.12
+  boot.kernelPatches = lib.mkDefault [{
+    name = "xps17-9710-fix-sound";
+    patch = null;
+    extraConfig = ''
+      SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES y
```

Comment:
How does this option fix sound? https://cateee.net/lkddb/web-lkddb/SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES.html

Reply:
Ok. get it. It's a dependency for `SND_SOC_INTEL_SOUNDWIRE_SOF_MACH`.
Can you make a PR for nixpkgs to include both options in nixpkgs master?

Reply:
Add it to `nixpkgs`, like here? https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/common-config.nix#L290

Reply:
Yes.

Reply:
@Mic92 I've created and tested a nixpkgs pr locally
https://github.com/NixOS/nixpkgs/pull/154220